graphql/server.go

```diff
@@ -500,6 +518,73 @@ func CreateJSONSocketWithMutationSchema(ctx context.Context, socket JSONSocket,
 	}
 }
 
+type noopSubscriptionLogger struct{}
+
+func (l *noopSubscriptionLogger) Subscribe(ctx context.Context, id string, tags map[string]string) {}
+func (l *noopSubscriptionLogger) Unsubscribe(ctx context.Context, id string)                       {}
+
+type noopGraphqlLogger struct{}
+
+func (l *noopGraphqlLogger) StartExecution(ctx context.Context, tags map[string]string, initial bool) {
+}
+func (l *noopGraphqlLogger) FinishExecution(ctx context.Context, tags map[string]string, delay time.Duration) {
+}
+func (l *noopGraphqlLogger) Error(ctx context.Context, err error, tags map[string]string) {}
+
+type ConnectionOption func(*conn)
+
+func CreateConnection(ctx context.Context, socket JSONSocket, schema *Schema, opts ...ConnectionOption) *conn {
+	c := &conn{
+		socket:        socket,
+		ctx:           ctx,
+		schema:        schema,
+		subscriptions: make(map[string]*reactive.Rerunner),
+	}
+	for _, opt := range opts {
+		opt(c)
+	}
+
+	if c.subscriptionLogger == nil {
```

Comment:
nso: save some lines by doing this above in struct literal and they will be simply override by functional options

Reply:
don